### PR TITLE
Limit the number of minutes and bytes per log file in LogCopier

### PR DESCRIPTION
**Pull request by zliu41**
@liyinan926 can you review?

### CODE REVIEW

File: gobblin-utility/src/main/java/gobblin/util/logs/LogCopier.java [hunk 1]
```diff
@@ -407,23 +443,49 @@ public LogCopier build() {
...
+      this.destLogDir = destLogFile.getParent();
+      this.destLogFileBaseName = FilenameUtils.removeExtension(destLogFile.getName());
+      this.destLogFileExtension = FilenameUtils.getExtension(destLogFile.getName());
+      this.currentDestLogFile = getCurrentDestLogFile();
+      this.watch.start();
+    }
+
+    private Path getCurrentDestLogFile() {
+      return new Path(this.destLogDir,
+          this.destLogFileBaseName + "." + System.currentTimeMillis() + "." + this.destLogFileExtension);
```
Comment (liyinan926) [inline gobblin-utility/src/main/java/gobblin/util/logs/LogCopier.java]: I suggest putting the timestamp at the end.

File: gobblin-utility/src/main/java/gobblin/util/logs/LogCopier.java [hunk 2]
```diff
@@ -407,23 +443,49 @@ public LogCopier build() {
...
+    private Path currentDestLogFile;
 
     // The task maintains the current source log file position itself
     private long currentPos = 0;
 
     LogCopyTask(Path srcLogFile, Path destLogFile) {
       this.srcLogFile = srcLogFile;
-      this.destLogFile = destLogFile;
+      this.destLogDir = destLogFile.getParent();
+      this.destLogFileBaseName = FilenameUtils.removeExtension(destLogFile.getName());
```
Comment (liyinan926) [inline gobblin-utility/src/main/java/gobblin/util/logs/LogCopier.java]: You can use Guava's `Files` utility class to replace `FilenameUtils`.

File: gobblin-utility/src/main/java/gobblin/util/logs/LogCopier.java [hunk 3]
```diff
@@ -407,23 +443,49 @@ public LogCopier build() {
...
+        LOGGER.debug(String.format("Copying changes from %s to %s", this.srcLogFile, this.currentDestLogFile));
+        copyChangesOfLogFile(srcFs.makeQualified(this.srcLogFile), destFs.makeQualified(this.currentDestLogFile));
       } catch (IOException ioe) {
-        LOGGER.error(String.format("Failed while copying logs from %s to %s", this.srcLogFile, this.destLogFile), ioe);
+        LOGGER.error(String.format("Failed while copying logs from %s to %s", this.srcLogFile, this.currentDestLogFile),
+            ioe);
+      }
+    }
+
+    private void getNextLogFileIfNeeded() throws IOException {
```
Comment (liyinan926) [inline gobblin-utility/src/main/java/gobblin/util/logs/LogCopier.java]: You will also need to reset `this.currentPos`.

File: gobblin-utility/src/main/java/gobblin/util/logs/LogCopier.java (hunk 1 above)
Comment (liyinan926) [inline gobblin-utility/src/main/java/gobblin/util/logs/LogCopier.java]: I think a better approach is to always name the current log file as normal like `GobblinApplicationMaster.stdout` and rename it to something like `GobblinApplicationMaster.stdout.<timestamp>` when necessary and re-create a new empty file named `GobblinApplicationMaster.stdout`.

File: gobblin-utility/src/main/java/gobblin/util/logs/LogCopier.java (hunk 3 above)
Comment (zliu41) [inline gobblin-utility/src/main/java/gobblin/util/logs/LogCopier.java]: currentPos is for the source file, not dest file.

**Comment by zliu41** [comment]
@liyinan926 updated 94026e1

**Comment by zliu41** [comment]
@liyinan926 updated 6c87cc8

**Comment by liyinan926** [comment]
LGTM.

**Pull request merged**